### PR TITLE
Completing exercise for module 1

### PR DESCRIPTION
**Pull request by ankanbanerjee22**
------------------
#### React JS Global Learning Program
-------------------
#### MODULE 1(Excercise)

- **Set up the environment**
Create a git repository on Github, Gitlab, Bitbucket or any other platform. Discuss with your mentor which platform is better.
Use create-react-app to bootstrap a new app. Commit the changes to repository. Install extensions mentioned in the lecture (react developer tools, etc.). 
Create components
To understand more how components work it's recommended to create a few interactive components.
Create the followong three components and render them all on one page, so that you can play around with them and test their look and behavior.

- **Counter**
A component that renders a numeric value and two buttons: one to decrement the value by 1, another to increment the value by 1.
The component should take an initial value in a property.
The component should be written using EcmaScript classes syntax, the `render` method should use `React.createElement` API (without JSX).

- **SearchForm**
Refer to the design prototype, implement a component that renders a search input and a button that triggers a new search.
The component should accept two properties:
Initial search query. Use the value to set the initial value for the input
A "onSearch" callback property. Call the callback property every time the user presses Enter when the input has focus or when the user clicks the Search button. Pass current input value in callback arguments.

- **GenreSelect**
Refer to the design prototype, implement a component that renders a list of movie genres with currently selected genre highlighted.
The component should accept three properties:
A list of genre names to display. Use the incoming list to render genre buttons.
A name of currently selected genre. Use the name to identify which button to highlight.
A "onSelect" callback property. Call the callback function when the user clicks on any genre button. Pass respective genre name to the callback arguments.

# Visuals

**Landing Page**
image

**Counter**
image

**Search Box**
image

**Movie Genre List**
image

### CODE REVIEW

File: module_1/first-react-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from 'react';
+import SearchBox from './components/SearchBox.js';
+import MovieGenre from './components/MovieGenre.js';
+import CounterLegacy from './components/CounterLegacy.js';
+
+
+
+
+
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/App.js]: Please remove the blank lines

File: module_1/first-react-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+function App() {
+
+  useEffect(() => {
+    // Initialize Materialize collapsible when the component mounts
+    const collapsibleElems = document.querySelectorAll('.collapsible');
+    window.M.Collapsible.init(collapsibleElems, {
+      accordion: false
+    });
+
+  }, []); // Empty dependency array ensures that this effect runs once after the initial render
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/App.js]: Please remove all the comments, you can keep comments locally but in the PR's better delete them.

File: module_1/first-react-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    //alert('Genre is: ' + genre);
+  };
+
+  const handleSearch = (query) => {
+    alert("Searching movies with name: " + query);
+  }
+
+
+
+
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/App.js]: Please remove empty lines

File: module_1/first-react-app/src/components/Counter.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        fontSize: '2.2rem', 
+        fontWeight: 'bold',
+        borderColor: 'black',
+        border: '2px solid black',
+        borderRadius: '20px', 
+    };
+
+    return (
+        <div className="container"> 
+
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/components/Counter.js]: Please remove blank line

File: module_1/first-react-app/src/components/Counter.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                    <button className="waves-effect red btn right-align" onClick={handleDecrement}>
+                        <i class="material-icons">remove</i>
+                    </button>
+                </div>
+                <div className="col s1"> 
+                    <div className="text-input center-align" style={chipStyle}> 
+                        {count}
+                    </div>
+                </div>
+                <div className="col s1 left-align" style={{ margin: '6px 20px' }}>
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/components/Counter.js]: to use inline styles is a bad practice, please create a separate css class for the styles

File: module_1/first-react-app/src/components/Counter.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                    <button className="waves-effect green btn" onClick={handleIncrement}>
+                        <i class="material-icons">remove</i>
+                    </button>
+                </div>
+            </div>
+        </div>
+    );
+};
+
+export default Counter;
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/components/Counter.js]: please add a blank line

File: module_1/first-react-app/src/components/MovieGenre.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+  return (
+    <div className="genre-list">
+      <div className="row">
+        <div className="col s12">
+          <ul className="tabs  z-depth-5" style={customTabsStyle} ref={tabsRef}>
+            {genres.map((genre) => (
+              <li key={genre} className="tab">
+                <a href={`#${genre}`} className={genre === selectedGenre ? 'active' : ''} onClick={() => onSelect(genre)}>
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/components/MovieGenre.js]: `#${genre}` - please put into a separate const
`genre === selectedGenre ? 'active' : '' ` - please put into a separate const

File: module_1/first-react-app/src/components/SearchBox.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        borderRadius: '15px'
+    };
+
+    const inputStyle = {
+        backgroundColor: '#263238',
+        borderRadius: '20px',
+        padding: '10px',
+        boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.5)',
+        width: '100%'
+    };
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/components/SearchBox.js]: Please move styles into a separate file

File: module_1/first-react-app/src/components/SearchBox.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+
+    return (
+        <>
+            <div className="row center-align">
+                <div className="col s12 m8 offset-m2">
+                    <div className="card medium" style={cardStyle}>
+                        <div className="card-content" style={overlayStyle}>
+                            <div className="row center-align">
+                                <div className="col s12 center-align">
+                                    <h3 style={{ color: '#eceff1' }}>Find Your Movie</h3>
```
Comment (andreyBalabukha) [inline module_1/first-react-app/src/components/SearchBox.js]: `color: '#eceff1'` - move to the styles
Find Your Movie - please move all the literals to a separate file, like literals.js 
for example: 
`export const FIND_YOUR_MOVIE = 'Find Your Movie'`

**Review by andreyBalabukha — CHANGES_REQUESTED** [review]

File: module_1/first-react-app/src/App.js (hunk 1 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/App.js]: Done in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/App.js (hunk 2 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/App.js]: Done in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/App.js (hunk 3 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/App.js]: Have removed in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/components/Counter.js (hunk 4 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/components/Counter.js]: cleaned up blank lines in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/components/Counter.js (hunk 5 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/components/Counter.js]: Have created seperate css files for each components in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/components/Counter.js (hunk 6 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/components/Counter.js]: Done in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/components/MovieGenre.js (hunk 7 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/components/MovieGenre.js]: Have introduced const for these in commit : c7a4e1e3bff35922c071c59fe3905a3da37fefef

File: module_1/first-react-app/src/components/SearchBox.js (hunk 8 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/components/SearchBox.js]: Have created separate css files in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

This was done mainly as I was trying out defining style in this way 😄

File: module_1/first-react-app/src/components/SearchBox.js (hunk 9 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/src/components/SearchBox.js]: Created new literals.js file in commit: c7a4e1e3bff35922c071c59fe3905a3da37fefef

**Review by andreyBalabukha — APPROVED** [review]

**Comment by andreyBalabukha** [comment]
Good job. Now you can merge your PR.

**Pull request merged**